photos/photos.py

```diff
@@ -155,56 +184,121 @@ def watermark_photo(image, watermark_text, watermark_image, watermark_image_size
     return image
 
 
-def resize_worker(orig, resized, spec, wm, wm_text, wm_img, wm_img_size):
+def rotate_image(img, exif_dict):
+
+    if "exif" in img.info and piexif.ImageIFD.Orientation in exif_dict["0th"]:
+        orientation = exif_dict["0th"].pop(piexif.ImageIFD.Orientation)
+        if orientation == 2:
+            img = img.transpose(Image.FLIP_LEFT_RIGHT)
+        elif orientation == 3:
+            img = img.rotate(180)
```

Comment:
The `transpose` and `rotate` perform the same operations. The only difference is the transpose operation implicitly sets the `expand` parameter to true. 

> The transpose(ROTATE) operations can also be performed identically with rotate() operations, provided the expand flag is true, to provide for the same changes to the image’s size.

http://pillow.readthedocs.io/en/3.3.x/handbook/tutorial.html#transposing-an-image
http://pillow.readthedocs.io/en/3.3.x/reference/Image.html
http://pillow.readthedocs.io/en/3.3.x/reference/Image.html#PIL.Image.Image.rotate

The code it replaces used the same operation to rotate the image 180.

```python
for tag, value in exif.items():
                    decoded = ExifTags.TAGS.get(tag, tag)
                    if decoded == 'Orientation':
                        if   value == 3: im = im.rotate(180)
                        elif value == 6: im = im.rotate(270)
                        elif value == 8: im = im.rotate(90)
                        break
```
https://github.com/bnice5000/pelican-plugins/blob/712607b3fc6190516b8278e2eebdc4d0c4a4f07e/photos/photos.py

The code I have take to replace it is a reproduction of the piexif example code.
http://piexif.readthedocs.io/en/latest/sample.html#rotate-image-by-exif-orientation

The past behavior and the documented behavior use the same code. We could change it, but the implicit expand could have unintended consequences.